test/TestUtil.js

```diff
@@ -30,6 +29,30 @@ exports.promiseLater = function (time, func) {
     }));
 };
 
+/**
+ * Returns rejection reason if rejected, otherwise returns a dummy error.
+ */
+exports.getRejectionReasonOrDummy = async function (object, asyncMethodName, ...params) {
+    try {
+        await object[asyncMethodName](...params);
```

Comment:
Instead of this, would it be more readable (both here and when we use it in tests) if we accept an async function?

So, it will be used as `await TestUtil.getRejectionReasonOrThrow(async () => await result.next())`